src/esp/assets/GenericMeshData.cpp

```diff
@@ -73,11 +81,49 @@ void GenericMeshData::setMeshData(Magnum::Trade::MeshData&& meshData) {
 
 void GenericMeshData::importAndSetMeshData(
     Magnum::Trade::AbstractImporter& importer,
-    int meshID) {
+    int meshID,
+    Corrade::PluginManager::Manager<Mn::Trade::AbstractSceneConverter>&
+        manager) {
   /* Guarantee mesh instance success */
   Cr::Containers::Optional<Mn::Trade::MeshData> mesh = importer.mesh(meshID);
   CORRADE_INTERNAL_ASSERT(mesh);
-  setMeshData(*std::move(mesh));
+
+  totalSourceIndices += mesh->indexCount();
+
+  if (targetMeshSimplificationFraction < 1.f) {
+    Mn::Containers::Pointer<Mn::Trade::MeshOptimizerSceneConverter>
+        meshoptimizer = Corrade::Containers::pointerCast<
+            Mn::Trade::MeshOptimizerSceneConverter>(
+            manager.loadAndInstantiate("MeshOptimizerSceneConverter"));
+
+    meshoptimizer->configuration().setValue("simplify", "true");
+    meshoptimizer->configuration().setValue(
+        "simplifyTargetIndexCountThreshold",
+        std::to_string(targetMeshSimplificationFraction));
+    constexpr float simplifyTargetError =
+        0.2;  // I don't think this field has any effect
+    meshoptimizer->configuration().setValue(
+        "simplifyTargetError", std::to_string(simplifyTargetError));
```

Comment:
...and same here (Configuration and Arguments use the same internals for type conversion):

```suggestion
        "simplifyTargetError", simplifyTargetError);
```